exist-core/src/main/java/org/exist/http/urlrewrite/XQueryURLRewrite.java

```diff
@@ -473,7 +473,9 @@ private ModelAndView getFromCache(final String url, final Subject user) throws E
 
         try (final DBBroker broker = pool.get(Optional.ofNullable(user))) {
 
-            model.getSourceInfo().source.validate(broker.getCurrentSubject(), Permission.EXECUTE);
+            if (model.getSourceInfo().source instanceof DBSource) {
+                ((DBSource) model.getSourceInfo().source).validate(Permission.EXECUTE);
+            }
```

Comment:
do we need a logging for the 'false' scenario?

Reply:
I don't believe so. This was done through inheritance before, the methods for every source sublass other than DBSource were just empty